system_tests/datastore.py

```diff
@@ -495,3 +514,59 @@ def test_failure_with_contention(self):
                 # transaction.
                 entity_in_txn[contention_prop_name] = u'inside'
                 txn.put(entity_in_txn)
+
+
+class TestDoctest(unittest.TestCase):
+
+    def _submodules(self):
+        pkg_iter = pkgutil.iter_modules(datastore.__path__)
+        result = []
+        for _, mod_name, ispkg in pkg_iter:
+            if mod_name == '_generated':
+                self.assertTrue(ispkg)
+            else:
+                self.assertFalse(ispkg)
+                result.append(mod_name)
+
+        self.assertNotIn('__init__', result)
+        return result
+
+    @staticmethod
+    def _add_section(index, mod_name, file_obj):
+        mod_part = 'datastore'
+        if mod_name != '__init__':
+            mod_part += '.' + mod_name
+        content = SPHINX_SECTION_TEMPLATE % (index, mod_part)
+        file_obj.write(content)
+
+    def _make_temp_docs(self):
+        docs_dir = tempfile.mkdtemp(prefix='datastore-')
+
+        conf_file = os.path.join(docs_dir, 'conf.py')
+
+        with open(conf_file, 'w') as file_obj:
+            file_obj.write(SPHINX_CONF)
+
+        index_file = os.path.join(docs_dir, 'contents.rst')
+        datastore_modules = self._submodules()
+        with open(index_file, 'w') as file_obj:
+            self._add_section(0, '__init__', file_obj)
+            for index, datastore_module in enumerate(datastore_modules):
+                self._add_section(index + 1, datastore_module, file_obj)
+
+        return docs_dir
+
+    def test_it(self):
+        from sphinx import application
+
+        docs_dir = self._make_temp_docs()
```

Comment:
@tseaver It's worth noting that this will only pick up doctests in source files. I think this is a fine start, and have some ideas how we can have automated checks that make sure we don't miss any tests.